scvi/core/models/rnamixin.py

```diff
@@ -104,43 +109,48 @@ def get_normalized_expression(
             model_fn = self.model.get_sample_scale
             scaling = library_size
 
-        exprs = []
-        for tensors in scdl:
-            x = tensors[_CONSTANTS.X_KEY]
-            batch_idx = tensors[_CONSTANTS.BATCH_KEY]
-            labels = tensors[_CONSTANTS.LABELS_KEY]
-            exprs += [
-                np.array(
-                    (
-                        model_fn(
-                            x,
-                            batch_index=batch_idx,
-                            y=labels,
-                            n_samples=n_samples,
-                            transform_batch=transform_batch,
-                        )[..., gene_mask]
-                        * scaling
-                    ).cpu()
-                )
-            ]
+        all_exprs = []
+        for batch in transform_batch:
```

Comment:
this loop has to go inside the tensors loop, see totalvi normalized expression code